### DISCUSSION TITLE
2023-11-06 회의

### DISCUSSION THREAD

**Discussion by chanhihi** [jiphyeonjeon-42/frontend#586] [meeting]
#582 

[오늘의 라이브 쉐어](https://prod.liveshare.vsengsaas.visualstudio.com/join?FE8A1063890ACD58622D0ED7711C7FF794B6)

[ERD](https://www.erdcloud.com/d/6rZ4ae7Ft6CrK83RT)

**Comment by chanhihi** [comment]
# 집현전 프론트엔드 온보딩 1단계

# Javascript
## 공통 조건

- 온보딩 프로젝트는 개인 계정으로 집현전 브랜치 전략(이슈 -> 브랜치 생성)을 기반으로 진행하고 PR로 제출합니다.
    - 이슈 제목은 [intra ID]
- [github] / [Commits Convention]은 집현전 Frontend Discussions 에 있는 자료를 적극 반영합니다.
- 기본 기능 외 추가 기능, 디자인 구현은 자유입니다.
- 최종 제출품에는 README 작성이 되있어야 합니다.([예시](https://github.com/jiphyeonjeon-42/frontend))


### todo list 만들기 Mandatory

- (필수) Javascript, HTML, CSS
- (필수) todo 생성(Create), 조회(Read), 삭제(Delete) 기능 구현하기 
- (필수) 개별 todo 완료 상태 체크 기능 구현하기
- (필수) todo 통계 (미완료 todo 개수 / 전체 todo 개수)
- (필수) 필터링 기능 구현 (전체 보기 / 미완료만 보기 / 완료만 보기)
- (필수) 로컬스토리지에 저장
  - 새로고침을 해도 이전에 추가한 todo를 확인할 수 있습니다

## todo list 만들기 Bonus
- (선택) 예쁘게 꾸미기
- (선택) todo 수정(Update) 기능 구현하기
- (선택) 전체 todo 완료 처리 기능 구현하기
- (선택) 무료로 배포하기 (Vercel, Github Pages, AWS 등)
- (선택) 사용자 인터렉션에 따른 디자인 변경(CSS)
  - 완료된 todo와 미완료 todo 구별하기
  - 모든 todo가 완료되었을 경우 통계 색상 변경 등
- (선택) Clean Code에 대한 고민
  - 중복된 로직을 함수로 재사용
  - 함수명, 변수명 (유의미한 이름 사용 지향)

## 참고
- [MDN](https://developer.mozilla.org/ko/)
- [모던 JavaScript 튜토리얼](https://ko.javascript.info/)
- Google!



[github]: https://github.com/jiphyeonjeon-42/frontend
[Commits Convention]: https://github.com/jiphyeonjeon-42/frontend/discussions/447
[intra ID]: https://github.com/jiphyeonjeon-42/jiphyeonjeon-42-onboarding-fe-01/issues

**Comment by chanhihi** [comment]
# 집현전 프론트엔드 온보딩 2단계

# React + Typescript
## 공통 조건

- 온보딩 프로젝트는 개인 계정으로 집현전 브랜치 전략(이슈 -> 브랜치 생성)을 기반으로 진행하고 PR로 제출합니다.
    - 이슈 제목은 [intra ID]
- [github] / [Commits Convention]은 집현전 Frontend Discussions 에 있는 자료를 적극 반영합니다.
- 기본 기능 외 추가 기능, 디자인 구현은 자유입니다.
- 최종 제출품에는 README 작성이 되있어야 합니다.([예시](https://github.com/jiphyeonjeon-42/frontend))


  
## todo list 만들기 Mandatory
- (필수) React (pnpm + vite + SWC), TypeScript, CSS
  - functional component 및 hooks 활용
  - 상태관리 라이브러리 사용금지
- (필수) eslint, prettier

### [온보딩 1단계]와 동일
- (필수) todo 생성(Create), 조회(Read), 삭제(Delete), **수정(Update)** 기능 구현하기 
- (필수) 개별 todo 완료 상태 체크 기능 구현하기
- (필수) todo 통계 (미완료 todo 개수 / 전체 todo 개수)
- (필수) 필터링 기능 구현 (전체 보기 / 미완료만 보기 / 완료만 보기)
- (필수) 로컬스토리지에 저장
  - 새로고침을 해도 이전에 추가한 todo를 확인할 수 있습니다.

## todo list 만들기 Bonus
- (선택) 예쁘게 꾸미기
- (선택) 전체 todo 완료 처리 기능 구현하기
- (선택) 무료로 배포하기 (Vercel, Github Pages, AWS 등)
- (선택) 사용자 인터렉션에 따른 디자인 변경(CSS)
  - 완료된 todo와 미완료 todo 구별하기
  - 모든 todo가 완료되었을 경우 통계 색상 변경 등
- (선택) Clean Code에 대한 고민
  - 중복된 로직을 함수로 재사용
  - 함수명, 변수명 (유의미한 이름 사용 지향)
- (선택) 랜더링 최적화


## 참고
- Google!
- [MDN](https://developer.mozilla.org/ko/)
- [모던 JavaScript 튜토리얼](https://ko.javascript.info/)
- [React](https://react.dev/)
- [React vite](https://vitejs.dev/guide/)
- [React](https://reactjs.org/docs/getting-started.html)
- [TypeScript](https://www.typescriptlang.org/docs/handbook/intro.html)
- [CSS](https://developer.mozilla.org/ko/docs/Web/CSS)
- [eslint](https://eslint.org/docs/user-guide/getting-started)
- [prettier](https://prettier.io/docs/en/index.html)


[github]: https://github.com/jiphyeonjeon-42/frontend
[Commits Convention]: https://github.com/jiphyeonjeon-42/frontend/discussions/447
[intra ID]: https://github.com/jiphyeonjeon-42/jiphyeonjeon-42-onboarding-fe-01/issues
[온보딩 1단계]: https://github.com/jiphyeonjeon-42/jiphyeonjeon-42-onboarding-fe-01

**Comment by mink97** [comment]
[타입재정의 live share](https://prod.liveshare.vsengsaas.visualstudio.com/join?B69C05DBB9E03D94DA469A8F2FD8817CB7F7)

**Reply by not-using** [reply]
<img width="802" alt="image" src="https://github.com/jiphyeonjeon-42/frontend/assets/74622889/17bf11da-246b-4de1-8817-3323adbe41ac">

**Comment by chanhihi** [comment]
## 진행하면서 고민한 내용
